packages/app/src/components/uiBlocks/Textfield.vue

```diff
@@ -0,0 +1,45 @@
+<template>
+  <div class="max-w-max flex flex-col">
+    <!-- label -->
+    <p v-if="label" class="font-bold pl-4">{{ label }}</p>
+    <!-- textfield with icon-->
+    <div class="border-2 border-yellow-400 px-4 py-1 rounded-full focus:ring-2 focus:ring-yellow-400 focus:ring-opacity-25 shadow-lg">
+      <span v-if="icon" class="mr-2" :class="['fas', `fa-${icon}`]" />
+      <input
+        class="focus: outline-none bg-transparent placeholder-opacity-50"
+        :placeholder="placeholder"
+        type="text"
+        :value="modelValue"
+        @input="$emit('update:modelValue', $event.target.value)"
+      />
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+import { defineComponent } from 'vue';
+
+export default defineComponent({
+  props: {
+    placeholder: {
+      type: String,
+      required: true,
+    },
+    modelValue: {
```

Comment:
what does this do?
I would suggest changing the name, just from plain reading the sourcecode I couldn't understand the function of this.
Is this the value entered by a user into the textfield?
How about

```suggestion
    inputvalue: {
```
or

```suggestion
    input: {
```

Reply:
The v-model command simly replace the command 'v-model' with the text v-bind:value="searchText"
v-on:input="searchText = $event.target.value". It gives a prop an wants to listen on it's change.
further reading on:
https://v3.vuejs.org/guide/component-custom-events.html#handling-v-model-modifiers

Reply:
Do you think a small comment might be good here?

Reply:
Not quite sure. This is the standard to read values from child components with vue 3 (to my knowlege). So my comment would just say, that it has to be like this.. no quite usefull. A link to the source as a comment might be outdated quickly.